File: C&Python/DFS/N과 M(1).py
check = [0 for i in range(9)]
v = [0 for i in range(9)]

def solution(n,m,index):
    if m == index:
        for i in range(m):
            print(v[i], end=' ')
        print(end='\n')
        return;

    for i in range(1,n + 1,1):
        if check[i] != 1:
            check[i] = 1
            v[index] = i
            solution(n,m,index +1)
            check[i] = 0


n, m = map(int, input().split())


solution(n,m,0)

File: C#/백준/N과 M(1).cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution
{
    static int N, M;
    static int[] check = new int[10];
    static List<string> answer = new List<string>();
    public static void DFS(int cnt)
    {
        // exit conditon
        if (cnt == M)
        {
            Console.WriteLine(string.Join(' ', answer));
            return;
        }


        for (int i = 1; i <= N; i++)
        {
            // visit
            if (check[i] == 0)
            {
                check[i] = 1;
                answer.Add(i.ToString());
                DFS(cnt + 1);
                answer.RemoveAt(answer.Count - 1);
                check[i] = 0;
            }
        }
    }

    public static void Main()
    {
        string input = Console.ReadLine();
        string[] sinput = input.Split(' ');

        for (int i = 0; i < 10; i++)
            check[i] = 0;

        N = int.Parse(sinput[0]);
        M = int.Parse(sinput[1]);
        DFS(0);
    }
}